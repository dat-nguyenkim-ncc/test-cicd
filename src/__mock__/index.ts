import { TagGroupType } from './../types/index.d'
import {
  CompanyBusiness,
  CompanyDetail,
  CompanyFinancials,
  CompanyOverview,
  CompanyPeople,
  MappingSummary,
  TagData,
  TagDataParent,
} from '../types'
import { EnumAttachmentType, EnumCompanyTypeSector } from '../types/enums'

const getRandomId = (): string => {
  return (~~(Math.random() * 999999) + new Date().valueOf()).toString()
}

const companyTypes = Object.values(EnumCompanyTypeSector)
const suffix = ['Restaurant', 'Limited', 'UK', 'US', 'Bank']
export const mockCompanyId = getRandomId()
export const companyDetails = (): CompanyDetail => ({
  companyId: mockCompanyId,
  companyName: `Monzo ${suffix[~~(Math.random() * suffix.length)]}`,
  countryCode: 'GB',
  countryName: 'United Kingdom',
  url: 'https://monzo.com',
  companyTypes: [companyTypes[~~(Math.random() * companyTypes.length)]],
})

export const MockMonzoDetails = companyDetails()

export const mappingSummary = [
  {
    title: 'CATEGORIES',
    items: [
      { label: 'Fintech', parent: [] },
      { label: 'Retail Accounts' },
      { label: 'Accounts & Savings' },
      { label: 'Digital Banking' },
    ],
    type: EnumCompanyTypeSector.FIN,
  },
  {
    title: 'SECTORS',
    items: [{ label: 'No sector mapped', parent: [] }],
    type: EnumCompanyTypeSector.FIN,
  },
] as MappingSummary[]

export const companySources = [
  {
    company: MockMonzoDetails,
    source: {
      label: 'Source: XPTO',
      default: true,
    },
  },
  {
    company: companyDetails(),
    source: {
      label: 'Manual',
    },
  },
]

export const companyOverview = (): CompanyOverview => ({
  id: getRandomId(),
  source: 'CRB',
  sources: ['CRB', 'DR', 'MANUAL'],
  companyName: `Monzo ${suffix[~~(Math.random() * suffix.length)]}`,
  companyLocation: [
    {
      id: '123456',
      address: 'Broadwalk House<br>5 Appold Street<br>London<br>EC2A 2AG',
      postalCode: '94103',
      location: {
        region: 'EMEA',
        city: 'San Francisco',
        country: 'United States',
      },
      isHeadQuarter: true,
      selfDeclared: true,
      expandStatus: 'Following',
      source: 'bcg',
    },
  ],
  description: 'Monzo is a fintech challenger bank.',
  companyType: 'Startup',
  foundedYear: 2012,
  status: 'The company status',
  expandStatus: 'Following',
  expandStatusId: '1',
  closedDate: new Date(2012, 11, 1).toISOString(),
  ftse: 1000,
  logoUrl: 'https://monzo.com/logo.svg',
  contactEmail: 'monzo@monzo.com',
  phoneNumber: '0123456789',
  tags: null,
  otherNames: null,
  lastFundingType: null,
  logo_bucket_url: '',
  hashed_image: '',
  attachments: [
    {
      name: 'File nam',
      description: 'Broadwalk House<br>5 Appold Street<br>London<br>EC2A 2AG',
      type: EnumAttachmentType.FCT_PROFILE,
      url_attachment: '',
      expandStatus: 'Following',
      selfDeclared: true,
      date_created: '2021-03-30 10:43:33',
    },
  ],
  aliases: [
    {
      alias_id: '200354bcg1',
      company_alias: 'name-edit',
      expand_status_id: 1,
      selfDeclared: true,
      source: 'bcg',
    },
    {
      alias_id: '200354bcg2',
      company_alias: 'test1',
      expand_status_id: 1,
      selfDeclared: true,
      source: 'bcg',
    },
  ],
})

export const companyFinancials = (): CompanyFinancials => ({
  valuation: {
    value: 23123123,
    currency: 'USD',
  },
  fundingTotal: {
    value: 2312312312,
    currency: 'USD',
  },
  companyStage: 'Series E+',
  lastFundingDate: {
    value: 1231233,
    currency: 'USD',
  },
  equityFundingTotal: {
    value: 1231233,
    currency: 'USD',
  },
  leadInvestor: 'Mussum Cassilds-Cepacol',
  lastFundingAmount: {
    value: 123123123,
    currency: 'USD',
  },
  fundingRounds: [
    {
      id: '1',
      date: new Date(2012, 11, 1).toISOString(),
      quarter: 'Q2',
      comment: 'This is comment',
      roundTypes: ['Equity Financing, series E+'],
      expandRound1: 'Equity Financing',
      originExpandRound1: 'Equity Financing',
      valuation: {
        value: 123123123,
        currency: 'USD',
      },
      investment: {
        value: 123123123,
        currency: 'USD',
      },
      investors: {
        lead: 'Accel Partners, VC',
        other:
          'Goodwater Capital, VC Orange, Corporate Passion Capital, VC Reference Capital (formerly Genevest), VC Stripe, Tech Thrive Capital, VC Vanderbilt University, Other Y Combinator, Accelerator',
      },
    },
    {
      id: '2',
      date: new Date(2012, 11, 1).toISOString(),
      quarter: 'Q2',
      comment: 'This is comment',
      roundTypes: ['Equity Financing, series E+'],
      expandRound1: 'Equity Financing',
      originExpandRound1: 'Equity Financing',
      valuation: {
        value: 123123123,
        currency: 'USD',
      },
      investment: {
        value: 123123123,
        currency: 'USD',
      },
      investors: {
        lead: 'Accel Partners, VC',
      },
    },
    {
      id: '3',
      date: new Date(2012, 11, 1).toISOString(),
      quarter: 'Q2',
      comment: 'This is comment',
      roundTypes: ['Equity Financing, series E+'],
      expandRound1: 'Equity Financing',
      originExpandRound1: 'Equity Financing',
      valuation: {
        value: 123123123,
        currency: 'USD',
      },
      investment: {
        value: 123123123,
        currency: 'USD',
      },
      investors: {
        lead: 'Accel Partners, VC',
      },
    },
  ],
  isExternalViewDetail: true,
})

export const companyBusiness = (): CompanyBusiness => ({
  productsServices: ['Offers a current account and debit card '],
  businessRevenue: ['Service type: B2C'],
  targetClients: ['Retail customers'],
  partnerships: [
    '**Deliveroo:** Partnered with Deliveroo in Sep-2017 to allow users to split the bill on orders via its Monzo.me money request feature',
  ],
  vision: ['string'],
  keyMetrics: ['string'],
  differentiators: ['string'],
})

export const companyAcquisitions = () => []
export const companyIpos = () => []
export const companyPeople = (): CompanyPeople => ({ companyId: 145803 })

const tags = [
  'Capital Markets',
  'Corporate Banking',
  'Retail Banking',
  'SME Banking',
  'Support',
  'Technology',
  'Wealth Management',
]

const getRandomValueLabel = (index?: number) => {
  const tag = index ? tags[index] : tags[~~(Math.random() * tags.length)]
  return {
    label: tag,
    value: tag.toLowerCase().replace(' ', '_'),
  }
}

const getRandomBoolean = () => {
  return Math.random() * 10 > 5
}

export const TagMock = (
  index?: number,
  level: number = 0,
  parent: TagDataParent[] = []
): TagData => {
  const { value, label } = getRandomValueLabel(index)
  const id = getRandomId()
  const rowId = getRandomId()
  const isPriority = getRandomBoolean()
  return {
    id,
    rowId,
    value,
    label,
    isPriority,
    endpoint: level > 0 ? Math.random() > 0.5 : false,
    parent,
    description: 'Mock',
    children:
      level >= 5
        ? undefined
        : new Array(Math.max(3, ~~(Math.random() * tags.length)))
            .fill(null)
            .map((_a, index) =>
              TagMock(index, level + 1, [...parent, { id, rowId, label, value, isPriority }])
            ),
  }
}

export const ListRandomTags = (): TagData[] =>
  new Array(tags.length).fill(null).map((_a, index) => TagMock(index, 0))

const labels = [
  'Insurance Value Chain 1',
  'Insure Tech Tag',
  'Revenue model',
  'P&C Commercial Customers',
  'Product',
  'RegTech',
  'Regulation',
  'Report',
]

export const TagGroupMock = (): TagGroupType[] =>
  new Array(labels.length * 2).fill(null).map((_a, index) => {
    const id = getRandomId()
    const rowId = getRandomId()
    const label = labels[index % labels.length]
    const value = label.toLowerCase().split(' ').join('_')
    const isPriority = getRandomBoolean()
    return {
      id,
      rowId,
      label,
      value,
      isPriority,
      parent: [],
      children: new Array(Math.max(1, ~~(Math.random() * labels.length * 2)))
        .fill(null)
        .map((_b, index) => {
          const labelChild = labels[index % labels.length]
          return {
            label: labelChild,
            id: getRandomId(),
            rowId: getRandomId(),
            parent: [{ id, label, value }],
            isPriority,
            value: labelChild.toLowerCase().split(' ').join('_'),
          }
        }),
    }
  })
