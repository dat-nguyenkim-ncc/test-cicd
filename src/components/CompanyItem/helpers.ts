import { CELL_SIZE } from '../../utils/consts'

export const GRID = `
  ${CELL_SIZE} 
  [checkbox] ${CELL_SIZE} 
  ${CELL_SIZE} 
  [company] repeat(6, ${CELL_SIZE}) 
  [company-end]
  ${CELL_SIZE} 
  [url] repeat(8, ${CELL_SIZE}) 
  [url-end]
  ${CELL_SIZE} 
  [source] repeat(4, ${CELL_SIZE}) 
  [source-end]
  ${CELL_SIZE} 
  [icons] repeat(6, ${CELL_SIZE}) 
  [icons-end]
`

export const CONFIRM_CHANGE_REQUEST = ` 
[date] repeat(4, ${CELL_SIZE}) 
[date-end]
${CELL_SIZE} 
[user] repeat(5, ${CELL_SIZE}) 
[user-end]
${CELL_SIZE} 
[previous-value] repeat(6, ${CELL_SIZE}) 
[previous-value-end]
${CELL_SIZE} 
[new-value] repeat(6, ${CELL_SIZE}) 
[new-value-end]
${CELL_SIZE}
[input-source] repeat(4, ${CELL_SIZE}) 
[input-source-end]
${CELL_SIZE}
[self-declared] repeat(3, ${CELL_SIZE}) 
[self-declared-end]
${CELL_SIZE}
[reason] repeat(10, ${CELL_SIZE})
[Reason-end]`

export const OVERRIDES_GRID = `
  [date] repeat(4, ${CELL_SIZE}) 
  [date-end]
  ${CELL_SIZE} 
  [user] repeat(8, ${CELL_SIZE}) 
  [user-end]
  ${CELL_SIZE} 
  [value] repeat(6, ${CELL_SIZE}) 
  [value-end]
  ${CELL_SIZE}
  [inputSource] repeat(4, ${CELL_SIZE}) 
  [inputSource-end]
  ${CELL_SIZE} 
  [selfDeclared] repeat(3, ${CELL_SIZE}) 
  [selfDeclared-end]
  ${CELL_SIZE} 
  [reason] repeat(6, ${CELL_SIZE}) 
  [reason-end]
  ${CELL_SIZE}
`

export const CQ_GRID = `
[date] repeat(4, ${CELL_SIZE}) 
[date-end]
${CELL_SIZE} 
[user] repeat(8, ${CELL_SIZE}) 
[user-end]
${CELL_SIZE} 
[value] repeat(6, ${CELL_SIZE}) 
[value-end]
${CELL_SIZE}
[inputSource] repeat(4, ${CELL_SIZE}) 
[inputSource-end]
${CELL_SIZE} 
[selfDeclared] repeat(3, ${CELL_SIZE}) 
[selfDeclared-end]
${CELL_SIZE} 
[reason] repeat(6, ${CELL_SIZE}) 
[reason-end]
${CELL_SIZE}
[action] repeat(6, ${CELL_SIZE})
[action-end]
${CELL_SIZE}
`

export const COMPANY_GRID = `
  ${CELL_SIZE} 
  [checkbox] ${CELL_SIZE} 
  ${CELL_SIZE} 
  [company] repeat(6, ${CELL_SIZE}) 
  [company-end]
  ${CELL_SIZE} 
  [url] repeat(8, ${CELL_SIZE}) 
  [url-end]
  ${CELL_SIZE} 
  [source] repeat(3, ${CELL_SIZE}) 
  [source-end]
  ${CELL_SIZE} 
  [icons] repeat(7, ${CELL_SIZE}) 
  [icons-end]
`
