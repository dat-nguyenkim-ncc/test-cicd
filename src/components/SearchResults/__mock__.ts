export const data = {
  external: [
    {
      //replace property id by company_id
      //TODO: change all records
      company_id: '7b4545e6-a34d-47c2-853e-ff9eb40c18ef',
      source: 'CRB',
      companyDetails: {
        companyId: '9185a077-c260-ac40-2678-fef9e645591b',
        companyName: 'Monzo',
        url: 'https://monzo.com/',
        countryName: 'United Kingdom',
        countryCode: 'GB',
      },
    },
    {
      id: 'a67f6ebd-c2e4-4c15-8657-4d40cc7626f1',
      source: 'DR',
      companyDetails: {
        companyId: '74473',
        companyName: 'Monzo',
        url: 'https://monzo.com',
        countryName: 'United Kingdom',
        countryCode: 'GB',
      },
    },
    {
      id: '89c20324-1390-4ea2-acbc-7cc05a98db59',
      source: 'CRB',
      companyDetails: {
        companyId: '33164743-a09a-45cb-a160-a54ace4dc3da',
        companyName: 'Monzon Biomed',
        url: 'http://www.monzonbiomed.com/',
        countryName: 'Spain',
        countryCode: 'ES',
      },
    },
    {
      id: '807a6c17-a4ff-4b74-b9ef-fee94364f451',
      source: 'CRB',
      companyDetails: {
        companyId: '456754eb-67c6-98db-780a-327db4c05c8f',
        companyName: 'monzoom.com',
        url: 'https://www.monzoom.com',
        countryName: 'Austria',
        countryCode: 'AT',
      },
    },
    {
      id: '0511b523-c25b-443b-932f-4d501d5c0cd4',
      source: 'CRB',
      companyDetails: {
        companyId: '09c52685-0f60-4a6f-a5fc-5c51ae5bec3a',
        companyName: 'Monzoon Networks',
        url: 'http://hotspot.monzoon.net',
        countryName: 'Switzerland',
        countryCode: 'CH',
      },
    },
  ],
  internal: [
    {
      //replace property id by company_id
      //TODO: change all records
      company_id: '6',
      source: 'MANUAL',
      companyDetails: {
        companyId: '6',
        companyName: 'MonzoBCG',
        url: 'www.MonzoBCG.com',
        countryName: 'England',
        countryCode: null,
      },
    },
    {
      id: '127237',
      source: 'DR',
      companyDetails: {
        companyId: '127237',
        companyName: 'Monzo',
        url: 'https://monzo.com',
        countryName: 'United Kingdom',
        countryCode: 'GB',
      },
    },
    {
      id: '141132',
      source: 'MANUAL',
      companyDetails: {
        companyId: '141132',
        companyName: 'Monzo',
        url: 'https://monzo.com/',
        countryName: 'United Kingdom',
        countryCode: 'GB',
      },
    },
  ],
}
