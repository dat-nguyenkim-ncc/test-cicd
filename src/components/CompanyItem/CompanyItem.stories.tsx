import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'

import CompanyItem, { CompanyItemProps } from './CompanyItem'
import { companyDetails } from '../../__mock__'
import { Box } from 'theme-ui'
import { BrowserRouter } from 'react-router-dom'

export default {
  title: 'Search/CompanyItem',
  component: CompanyItem,
} as Meta

const Template: Story<CompanyItemProps> = args => (
  <Box sx={{ p: 6, backgroundColor: 'white' }}>
    <BrowserRouter>
      <CompanyItem {...args} />
    </BrowserRouter>
  </Box>
)

export const Listing = Template.bind({})
Listing.args = {
  companyDetails: companyDetails(),
  source: { label: 'Source: XPTO' },
  type: 'internal',
  onCheck: () => {
    Listing.args!.checked = !!!Listing.args!.checked
  },
} as CompanyItemProps

export const ChangeSource = Template.bind({})
ChangeSource.args = {
  companyDetails: companyDetails(),
  type: 'external',
  source: { label: 'Source: XPTO' },
} as CompanyItemProps
