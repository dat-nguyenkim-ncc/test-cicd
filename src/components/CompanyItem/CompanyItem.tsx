import React from 'react'
import { Flex, Grid } from 'theme-ui'
import { SearchResultItem, Source, ViewInterface } from '../../types'
import { Button, Checkbox, Link, Pill } from '../'
import { Paragraph } from '../primitives'
import { GRID } from './helpers'
import { SearchBlockType } from '../SearchResults/SearchResults'
import strings from '../../strings'
import { EnumExpandStatus, EnumExpandStatusId, Routes } from '../../types/enums'

export type CompanyItemProps = SearchResultItem &
  ViewInterface<{
    invertBg?: boolean
    checked?: boolean
    toBeRemoved?: boolean
    aggregated?: boolean
    onCheck?(id: string): void
    onRemove?(id: string): void
    type?: keyof typeof SearchBlockType
    isInDefaultSelected?: boolean
    isInReAggregate?: boolean
    columns?: string
    disabled?: boolean
  }>

const CompanyItem = ({
  companyDetails,
  source,
  checked,
  onCheck,
  onRemove,
  invertBg,
  aggregated,
  type = 'internal',
  toBeRemoved = false,
  sx,
  isInDefaultSelected,
  isInReAggregate,
  columns,
  disabled,
}: CompanyItemProps) => {
  const { companyList: copy } = strings

  const onCheckTick = () => {
    onCheck &&
      onCheck(
        isInReAggregate && isInDefaultSelected
          ? companyDetails.external_id || ''
          : companyDetails.companyId
      )
  }

  const onRemoveClick = () => {
    onRemove && onRemove(companyDetails.companyId)
  }

  const getConditionOfCheck = () => {
    if (isInReAggregate || isInDefaultSelected) {
      return checked
    }
    return !!(source && (source as Source).default)
  }

  const to =
    type === 'internal'
      ? Routes.COMPANY.replace(':id', companyDetails.companyId)
      : Routes.COMPANY_NEW.replace(':id', companyDetails.companyId).replace(
          ':source',
          source.toString()
        )

  return (
    <Flex sx={{ opacity: toBeRemoved ? 0.4 : 1, ...sx }}>
      <Grid
        gap={'1px'}
        columns={columns || GRID}
        sx={{
          alignItems: 'center',
          borderRadius: 10,
          backgroundColor: !aggregated ? (invertBg ? 'white' : 'gray03') : 'transparent',
          py: 2,
        }}
      >
        {onCheck && (
          <Checkbox
            disabled={disabled}
            onPress={onCheckTick}
            sx={{ gridColumn: 'checkbox' }}
            checked={checked}
          />
        )}

        <Link
          variant="company"
          sx={{
            whiteSpace: 'nowrap',
            overflow: 'hidden',
            textOverflow: 'ellipsis',
            gridColumn: 'company / company-end',
          }}
          to={to}
        >
          {companyDetails.companyName}
        </Link>

        {companyDetails.url && (
          <Paragraph
            sx={{
              whiteSpace: 'nowrap',
              overflow: 'hidden',
              textOverflow: 'ellipsis',
              gridColumn: 'url /url-end',
              '& > a': { color: 'black50' },
            }}
          >
            {companyDetails.url}
          </Paragraph>
        )}

        {source && (
          <Paragraph sx={{ gridColumn: 'source /source-end' }}>
            {typeof source === 'string' ? source : (source as Source).label}
          </Paragraph>
        )}
        <Flex sx={{ mr: 2, justifyContent: 'flex-end', gridColumn: 'icons / icons-end' }}>
          {getConditionOfCheck() && (
            <Pill sx={{ mr: 2 }} label={copy.defaultSource} variant="muted" />
          )}
          {companyDetails.expandStatusId === EnumExpandStatusId.DUPLICATED ? (
            <Pill sx={{ mr: 2 }} icon={EnumExpandStatus.DUPLICATED as any} variant="out" />
          ) : (
            companyDetails.primaryCategories?.map((t, index) => (
              <Pill key={index} sx={{ mr: 2 }} icon={t as any} />
            ))
          )}
          {companyDetails.countryCode && (
            <Pill alt={companyDetails.countryName} flag={companyDetails.countryCode} />
          )}
        </Flex>
      </Grid>
      {onRemove && (
        <Button sx={{ ml: 4 }} size="small" variant="black" icon="remove" onPress={onRemoveClick} />
      )}
    </Flex>
  )
}

export default CompanyItem
