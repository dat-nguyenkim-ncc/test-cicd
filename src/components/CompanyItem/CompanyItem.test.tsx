import * as React from 'react'
import { shallow } from 'enzyme'
import CompanyItem from './CompanyItem'
import { companyDetails } from '../../__mock__'

describe('CompanyItem', () => {
  test('matches snapshot', () => {
    const wrapper = shallow(
      <CompanyItem
        type="internal"
        companyDetails={companyDetails()}
        source={{ label: 'Source A-PXS' }}
        checked
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
