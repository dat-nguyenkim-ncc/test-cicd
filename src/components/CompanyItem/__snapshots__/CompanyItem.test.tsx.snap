// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyItem matches snapshot 1`] = `ShallowWrapper {}`;
